File: src/main/Java/boj_step21/boj27433.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;

public class boj27433 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new java.io.OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        writer.append(String.valueOf(factorial(n))).close();
    }
    private static long factorial (int n) {
        if (n <= 1) return 1;
        return n * factorial(n - 1);
    }
}

File: src/main/boj_step21/boj27433.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.math.BigInteger

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    w.write(factorial(n).toString())
    w.close()
}

private fun factorial(n: Int): BigInteger {
    if (n <= 1) return BigInteger.valueOf(1)
    return n.toBigInteger() * factorial(n - 1)
}